TicketsDemo.Domain/DefaultImplementations/TeaCoffeeBedPriceCalcStrategy.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Data.OptionsForCalculationPrice;
+using TicketsDemo.Data.Repositories;
+using TicketsDemo.Domain.DefaultImplementations.PriceCalculationStrategy;
+using TicketsDemo.Domain.Interfaces;
+
+namespace TicketsDemo.Domain.DefaultImplementations.TeaCoffeeBedPriceCalcStrategy
+{
+    public class TeaCoffeeBedPriceCalcStrategy : IPriceCalculationStrategy
+    {
+        private IRunRepository _runRepository;
+        private ITrainRepository _trainRepository;
+
+
+        public TeaCoffeeBedPriceCalcStrategy(IRunRepository runRepository, ITrainRepository trainRepository)
+        {
+            _runRepository = runRepository;
+            _trainRepository = trainRepository;
+        }
+
+        public List<PriceComponent> CalculatePrice(TeaCoffeeBedParametrs teaCoffeeBedParametrs)
+        {
+            var DefaultPriceStrategy = new DefaultPriceCalculationStrategy(_runRepository, _trainRepository);
```

Comment:
that's a violation of both DependencyInjection approach to dependencies management and DependencyInversion principle